format-config-v1.go

```diff
@@ -37,66 +44,104 @@ type formatConfigV1 struct {
 	XL      *xlFormat `json:"xl,omitempty"`
 }
 
-// FIXME: currently we don't check single exportPath which uses FS layer.
-
 // loadFormatXL - load XL format.json.
-func loadFormatXL(storage StorageAPI) (xl *xlFormat, err error) {
-	offset := int64(0)
-	r, err := storage.ReadFile(minioMetaBucket, formatConfigFile, offset)
-	if err != nil {
-		return nil, err
-	}
-	decoder := json.NewDecoder(r)
-	formatXL := formatConfigV1{}
-	err = decoder.Decode(&formatXL)
-	if err != nil {
-		return nil, err
-	}
-	if err = r.Close(); err != nil {
-		return nil, err
-	}
-	if formatXL.Version != "1" {
-		return nil, fmt.Errorf("Unsupported version of backend format [%s] found.", formatXL.Version)
-	}
-	if formatXL.Format != "xl" {
-		return nil, fmt.Errorf("Unsupported backend format [%s] found.", formatXL.Format)
+func loadFormatXL(disks ...string) (jbodMap map[string]formatConfigV1, err error) {
+	jbodMap = make(map[string]formatConfigV1, len(disks))
+	for _, disk := range disks {
+		formatConfigPath := filepath.Join(disk, minioMetaBucket, formatConfigFile)
+		r, err := os.Open(formatConfigPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return nil, err
+		}
+		decoder := json.NewDecoder(r)
+		formatXL := formatConfigV1{}
+		err = decoder.Decode(&formatXL)
+		if err != nil {
+			return nil, err
+		}
+		if err = r.Close(); err != nil {
+			return nil, err
+		}
+		// Validate format version and format type.
+		if formatXL.Version != "1" {
+			return nil, fmt.Errorf("Unsupported version of backend format [%s] found.", formatXL.Version)
+		}
+		if formatXL.Format != "xl" {
+			return nil, fmt.Errorf("Unsupported backend format [%s] found.", formatXL.Format)
+		}
+		// Save valid formats in a map.
+		jbodMap[disk] = formatXL
```

Comment:
This call is directly reading from the disk. It is supposed to create a boostrap stack based on disk type (network or local) and then load it.